### ISSUE TITLE
Where size  5 comes from in LlamaModelforCasualLM??

### ISSUE THREAD

**Issue by daehuikim** [huggingface/transformers#28992]
```
from transformers import (
    AutoModelForCausalLM,
    AutoTokenizer
)

model_name = "meta-llama/Llama-2-7b-chat-hf"
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelForCausalLM.from_pretrained(model_name)

input_text = "How are you?"
input_ids = tokenizer.encode(input_text, return_tensors="pt")
outputs = model(input_ids,output_hidden_states=True)

hidden_states = outputs.hidden_states
output_last = hidden_states[-1].detach()
print(output_last.shape)
```
Sorry for question here.
I am watching llama model codes for getting hidden states values from here(https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py)
When I try the code above, I can get the output like this
```
torch.Size([1, 5, 4096])
```
From here, I can understand ```1``` is batch size and ```4096```is size of max_position_embedding of this model.
Then What does ```5``` means from the hidden states of this model? 
(hidden states became logits once they go through ```lm_head``` module)
https://github.com/huggingface/transformers/blob/da20209dbc26a6a870a6e7be87faa657b571b7bc/src/transformers/models/llama/modeling_llama.py#L1155

Thanks for reading my question!

**Comment by amyeroberts** [comment]
Hi @daehuikim, thanks for raising an issue! 

This is a question best placed in our [forums](https://discuss.huggingface.co/). We try to reserve the github issues for feature requests and bug reports.

The value 5 comes from the input sequence length (the number of input tokens) i.e. `len(input_ids[0])`

**Comment by daehuikim** [comment]
@amyeroberts 
Thank you for your kindness!
